goas/atomic_asset_transfer/executor/tests/deposit_ptx.rs

```diff
@@ -0,0 +1,134 @@
+use std::collections::BTreeMap;
+
+use cl::{NoteWitness, NullifierNonce, NullifierSecret};
+use common::{BoundTx, StateWitness, Tx, ZoneMetadata, ZONE_CL_FUNDS_UNIT};
+use ledger::death_constraint::DeathProof;
+use rand_core::CryptoRngCore;
+
+fn zone_fund_note(value: u64, zone_meta: ZoneMetadata) -> cl::NoteWitness {
+    cl::NoteWitness {
+        value,
+        unit: *common::ZONE_CL_FUNDS_UNIT,
+        death_constraint: zone_meta.funds_vk,
+        state: zone_meta.id(),
+    }
+}
+
+fn zone_state_utxo(zone: &StateWitness, mut rng: impl CryptoRngCore) -> cl::OutputWitness {
+    cl::OutputWitness::public(
+        cl::NoteWitness {
+            value: 1,
+            unit: zone.zone_metadata.unit,
+            death_constraint: zone.zone_metadata.zone_vk,
+            state: zone.commit().0,
+        },
+        NullifierNonce::random(&mut rng),
+    )
+}
+
+#[test]
+fn test_deposit() {
+    let mut rng = rand::thread_rng();
+
+    let alice = 42;
+    let alice_sk = NullifierSecret::random(&mut rng);
+
+    let init_state = StateWitness {
+        balances: BTreeMap::new(),
+        included_txs: vec![],
+        zone_metadata: executor::zone_metadata("ZONE"),
+        nonce: [0; 32],
+    };
+
+    let zone_state_in = cl::InputWitness::public(zone_state_utxo(&init_state, &mut rng));
+
+    let deposit = common::Deposit {
+        to: alice,
+        amount: 78,
+    };
+
+    let end_state = init_state.clone().deposit(deposit).evolve_nonce();
+
+    let zone_state_out = cl::OutputWitness::public(
+        cl::NoteWitness {
+            state: end_state.commit().0,
+            ..zone_state_in.note
+        },
+        zone_state_in.evolved_nonce(),
+    );
+    let zone_fund_out = cl::OutputWitness::public(
+        zone_fund_note(78, init_state.zone_metadata),
+        NullifierNonce::from_bytes(end_state.nonce),
+    );
+
+    let mut alice_state = [0u8; 32];
+    alice_state[..4].copy_from_slice(&alice.to_le_bytes());
```

Comment:
I don't think we need this anymore? The user intent is responsible for ensuring the deposit goes to the correct account